src/gui/mzroll/mainwindow.cpp

```diff
@@ -3513,30 +3513,29 @@ void MainWindow::setClipboardToGroup(PeakGroup* group) {
 	clipboard->setText(groupTextExport(group));
 }
 
-void MainWindow::showFragmentationScans(float pmz) {
-
-	if (!fragPanel || fragPanel->isVisible() == false)
-		return;
+void MainWindow::showFragmentationScans(float pmz)
+{
+    if (!fragPanel || fragPanel->isVisible() == false)
+        return;
 	
-	MassCutoff *massCutoff = getUserMassCutoff();
+    MassCutoff *massCutoff = getUserMassCutoff();
 
-	if (samples.size() <= 0)
-		return;
-	fragPanel->clearTree();
-	for (auto sample : samples) {
-		for (auto scan : sample->scans) {
-			if (scan->mslevel < 2) continue;
-			if (massCutoffDist(scan->precursorMz,
-							   pmz,
-							   massCutoff) > massCutoff->getMassCutoff()) {
-				continue;
-			}
-			fragPanel->addScanItem(scan);
-		}
-	}
+    if (samples.size() <= 0)
+        return;
+    fragPanel->clearTree();
+    for (auto sample : samples) {
+        for (auto scan : sample->scans) {
+	        if (scan->mslevel < 2) continue;
+	        if (massCutoffDist(scan->precursorMz,
+                               pmz,
+                               massCutoff) > massCutoff->getMassCutoff()) {
```

Comment:
Again, `massCutoff` will probably not be needed after this point. Lets free the allocated memory.

Reply:
This one is a local pointer to a global variable, new memory has not been allocated.